File: scripts/int32_publisher.py
#!/usr/bin/env python
import rospy
from std_msgs.msg import Int32
import random

def publisher():
	pub = rospy.Publisher('integer',Int32, queue_size  =100)
	ros.init_node("int32_publisher", anonymous=True)
	rate = rospy.Rate(10)

	while not rospy.is_shutdown():
		var = random.randint(0,100)
		rospy.loginfo("I sent: %d", var)
		pub.publish(var)
		rte.sleep()

if __name__ == '__main__':
	try:
		publisher()
	except rospy.ROSInterruptException():
		subc

File: ros_basics/src/int32_publisher.cpp
#include "ros/ros.h"
#include "std_msgs/Int32.h"

#include <iostream>

int main(int argc, char **argv)
{
	ros::init(argc, argv, "int32_publisher");
	ros::NodeHandle nh;

	ros::Publisher pub = nh.advertise<std_msgs::Int32>("/integer",10);
	ros::Rate loop_rate(2);

	int count = 0;

	while(ros::ok())
	{
		std_msgs::Int32 msg;
		msg.data = count;
		ROS_INFO("Publisher sent: %d",msg.data);
		pub.publish(msg);

		ros::spinOnce();
		loop_rate.sleep();
		++count;
	}
	return 0;
}